pkg/scheduler/scheduler.go

```diff
@@ -670,6 +676,19 @@ func (sched *Scheduler) scheduleOne() {
 	}()
 }
 
+type podConditionUpdaterImpl struct {
+	Client clientset.Interface
+}
+
+func (p *podConditionUpdaterImpl) update(pod *v1.Pod, condition *v1.PodCondition) error {
```

Comment:
can't this be a private function of the Scheduler? why do we need the ```PodConditionUpdater``` interface? tests seem to use one that returns always ok.

Reply:
We lose the clientset somewhere in the pipeline. I think we can revisit them once we get rid of the Configurator and we can pass a fake clientset to the scheduler.

Reply:
We can add it to the Scheduler struct, and also use it for the "pod preemptor" as well since that interface also seems unnecessary.

Reply:
sgtm